src/libostree/ostree-repo-pull.c

```diff
@@ -3892,6 +3923,16 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
 
   g_hash_table_unref (requested_refs_to_fetch);
   requested_refs_to_fetch = g_steal_pointer (&updated_requested_refs_to_fetch);
+  if (g_hash_table_size (requested_refs_to_fetch) == 1)
+    {
+      GLNX_HASH_TABLE_FOREACH_KV (requested_refs_to_fetch,
```

Comment:
Nit: you can just use `GLNX_HASH_TABLE_FOREACH` here since we don't need the value.